### PR TITLE
github-ci: add workflow_dispatch

### PR DESCRIPTION
**Pull request by jasonish**
A workflow dispatch allows us to manually a trigger a workflow with
arguments. This dispatch allows us to use the "gh" cli command to
trigger a workflow run with our libhtp/su/sv branch and repo variables
set. For example:

  gh run builds.yml -f SV_REPO=jasonish/suricata-verify -f SV_BRANCH=pr/10

### CODE REVIEW

**Review by catenacyber — APPROVED** [review]
Looks good to me, how did you test it ?

**Comment by jasonish** [comment]
> Looks good to me, how did you test it ?

Push it to my own master then running something like:
```
gh -R jasonish/suricata workflow run builds.yml -f SV_REPO=jasonish/suricata-verify -f SV_BRANCH=pr-staging/1
```

**Comment by catenacyber** [comment]
Could you rebase, so that I approve again, to see if firing CODEOWNERS fixed the approved PR filter ?

**Comment by jasonish** [comment]
> Could you rebase, so that I approve again, to see if firing CODEOWNERS fixed the approved PR filter ?

Done.

**Review by catenacyber — APPROVED** [review]
Rebase is good

**Comment by victorjulien** [comment]
Merged in #8951, thanks!

**Pull request closed without merging**
